### PR TITLE
Allow to create a heightmap collision shape from an image

### PR DESCRIPTION
**Pull request by Zylann**
Before I had to manually convert all pixels in GDScript to a `PoolRealArray`, so I added the possibility to pass an image, which is faster and simpler.

### CODE REVIEW

**Comment by akien-mga** [comment]
CC @AndreaCatania

**Comment by akien-mga** [comment]
Thanks!

File: modules/bullet/shape_bullet.cpp [hunk 1]
```diff
@@ -438,7 +438,47 @@ void HeightMapShapeBullet::set_data(const Variant &p_data) {
...
+		ERR_FAIL_COND(l_image.is_null());
+
+		// Float is the only common format between Godot and Bullet that can be used for decent collision.
+		// (Int16 would be nice too but we still don't have it)
+		// We could convert here automatically but it's better to not be intrusive and let the caller do it if necessary.
+		ERR_FAIL_COND(l_image->get_format() != Image::FORMAT_RF);
+
+		PoolByteArray im_data = l_image->get_data();
+
+		l_heights.resize(l_image->get_width() * l_image->get_width());
```
Comment (rodolforg) [inline modules/bullet/shape_bullet.cpp]: Is this right? Should it be l_image->get_width() * l_image->get_height() ?

Comment (Zylann) [reply modules/bullet/shape_bullet.cpp]: That's a mistake, it should be width * height.

**Pull request merged**